src/framework/mlt_chain.c

```diff
@@ -393,6 +393,37 @@ void mlt_chain_close( mlt_chain self )
 	}
 }
 
+/** Attach normalizer links.
+ *
+ * \public \memberof mlt_chain_s
+ * \param self a chain
+ */
+
+extern void mlt_chain_attach_normalisers( mlt_chain self )
+{
+	if ( self )
+	{
+		int i = 0;
+		mlt_chain_base *base = self->local;
+		if ( base->link_count && mlt_properties_get_int( MLT_LINK_PROPERTIES( base->links[ i ] ), "_loader" ) )
+		{
+			mlt_log_warning( MLT_CHAIN_SERVICE(self), "Chain already has normalizers\n");
+			return;
+		}
+		mlt_link resample = mlt_factory_link( "swresample", NULL );
+		if ( !resample )
+		{
+			resample = mlt_factory_link( "resample", NULL );
```

Comment:
I wonder if there should be an ini file for this. Could there ever be like like a speed link added for every chain in order to do a better trick playback? Or a rate-doubling deinterlace filter?

Reply:
That is a good idea. I aspire to make a frame doubling deinterlacer.

Reply:
This is done now.